applications/nrf_desktop/src/modules/hid_state.c

```diff
@@ -96,11 +96,15 @@ struct report_data {
 	struct eventq eventq;
 	struct axis_data axes;
 	bool update_needed;
+	struct report_state *linked_rs;
```

Comment:
Then the `report_send` function could be used with the `report_state` instead of `report_data`.
I think this would be cleaner and easier to maintain.

Reply:
Discussed offline. Brief explanation - the change is done because report_data will no longer be directly linked with report_state (as it was so far). Instead the link will be dynamic. E.g. keyboard report_data can be used either by report_state associated with keyboard report or boot keyboard report.
So now when the button is pressed we cannot tell to which report it will go. We cannot then look for report_state. We have to look for report data it belongs to and check if there is any report state linked to this data at the moment.

Reply:
Discussed offline. The proposed solution is more universal one. Lets leave it as it is.

Reply:
BTW - personally I would use different enums for indexing report_data and report_state as this would simplify understanding of how data propagates. I am 90% sure this should be done but I don't want to do it yet in this PR.